### ISSUE TITLE
Exception on typing in TS file: System.ArgumentException Message=The specified SnapshotPoint or SnapshotSpan  is on a different ITextSnapshot than this SnapshotSpan.

### ISSUE THREAD

**Issue by minestarks** [mono/monodevelop#4934]
- Build Monodevelop latest `master` commit cde0652cfd08a61b365578bcddaab29b71f32786
- Build WebToolingAddin latest `master` commit 4ff1d20771f0775274ecf55aadffb918ad2e6345

Repro:
1. Clone https://github.com/Microsoft/TypeScript
2. Open [src/compiler/core.ts](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/core.ts)
3. Place cursor after a function declaration, hit Enter a couple of times and start typing, for example "`export function`"...

Exception:
```
System.ArgumentException
  HResult=0x80070057
  Message=The specified SnapshotPoint or SnapshotSpan  is on a different ITextSnapshot than this SnapshotSpan.
  Source=Microsoft.VisualStudio.Text.Data
  StackTrace:
   at Microsoft.VisualStudio.Text.SnapshotSpan.EnsureSnapshot(ITextSnapshot requestedSnapshot) in E:\A\_work\24\s\src\Editor\Text\Def\TextData\Model\SnapshotSpan.cs:line 491
```

Callstack:

``` 	Microsoft.VisualStudio.Text.Data.dll!Microsoft.VisualStudio.Text.SnapshotSpan.EnsureSnapshot(Microsoft.VisualStudio.Text.ITextSnapshot requestedSnapshot) Line 491	C#
 	Microsoft.VisualStudio.Text.Data.dll!Microsoft.VisualStudio.Text.SnapshotSpan.OverlapsWith(Microsoft.VisualStudio.Text.SnapshotSpan snapshotSpan) Line 303	C#
>	MonoDevelop.SourceEditor.dll!Mono.TextEditor.MdTextViewLineCollection.MdTextViewLine.GetNormalizedTextBounds(Microsoft.VisualStudio.Text.SnapshotSpan bufferSpan) Line 199	C#
 	MonoDevelop.SourceEditor.dll!Mono.TextEditor.MdTextViewLineCollection.GetNormalizedTextBounds(Microsoft.VisualStudio.Text.SnapshotSpan bufferSpan) Line 140	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.PopupAgent.PositionAndDisplay(System.Windows.Media.Geometry reservedSpace) Line 107	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SpaceReservationManager.PositionAndDisplay.AnonymousMethod__0() Line 206	C#
 	Microsoft.VisualStudio.Text.Implementation.dll!Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(object errorSource, System.Action call) Line 376	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SpaceReservationManager.PositionAndDisplay(System.Windows.Media.GeometryGroup reservedGeometry) Line 195	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SpaceReservationStack.Refresh() Line 96	C#
 	MonoDevelop.SourceEditor.dll!Mono.TextEditor.MonoTextEditor.QueueSpaceReservationStackRefresh.AnonymousMethod__111_0() Line 359	C#
 	MonoDevelop.Core.dll!MonoDevelop.Core.Runtime.RunInMainThread(System.Action action) Line 320	C#
 	MonoDevelop.SourceEditor.dll!Mono.TextEditor.MonoTextEditor.QueueSpaceReservationStackRefresh() Line 355	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SpaceReservationManager.ChangeAgents(Microsoft.VisualStudio.Text.Editor.ISpaceReservationAgent oldAgent, Microsoft.VisualStudio.Text.Editor.ISpaceReservationAgent newAgent) Line 146	C#
 	MonoDevelop.SourceEditor.dll!Microsoft.VisualStudio.Text.Editor.Implementation.SpaceReservationManager.AddAgent(Microsoft.VisualStudio.Text.Editor.ISpaceReservationAgent agent) Line 70	C#
 	MonoDevelop.CSharpBinding.dll!MonoDevelop.Ide.Completion.Presentation.RoslynCompletionPresenterSession.Open(Microsoft.VisualStudio.Text.ITrackingSpan triggerSpan, System.Collections.Generic.IList<Microsoft.CodeAnalysis.Completion.CompletionItem> items, Microsoft.CodeAnalysis.Completion.CompletionItem selectedItem, Microsoft.CodeAnalysis.Completion.CompletionItem suggestionModeItem, bool suggestionMode, bool isSoftSelected) Line 581	C#
 	MonoDevelop.CSharpBinding.dll!MonoDevelop.Ide.Completion.Presentation.RoslynCompletionPresenterSession.Microsoft.CodeAnalysis.Editor.ICompletionPresenterSession.PresentItems(Microsoft.VisualStudio.Text.ITrackingSpan triggerSpan, System.Collections.Generic.IList<Microsoft.CodeAnalysis.Completion.CompletionItem> items, Microsoft.CodeAnalysis.Completion.CompletionItem selectedItem, Microsoft.CodeAnalysis.Completion.CompletionItem suggestionModeItem, bool suggestionMode, bool isSoftSelected, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Completion.CompletionItemFilter> completionItemFilters, string filterText) Line 29	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Controller.OnModelUpdated(Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Model modelOpt) Line 154	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AbstractController<System.__Canon, System.__Canon, System.__Canon, System.__Canon>.Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.IController<System.__Canon>.OnModelUpdated(System.__Canon result) Line 77	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.ModelComputation<Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Model>.OnModelUpdated(Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Model result, bool updateController) Line 181	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.ModelComputation<Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Model>.ChainTaskAndNotifyControllerWhenFinished.AnonymousMethod__1(System.Threading.Tasks.Task[] tasks) Line 159	C#
 	mscorlib.dll!System.Threading.Tasks.TaskFactory<System.Threading.Tasks.VoidTaskResult>.ContinueWhenAllImpl.AnonymousMethod__57_1(System.Threading.Tasks.Task<System.Threading.Tasks.Task[]> completedTasks, object state) Line 1868	C#
 	mscorlib.dll!System.Threading.Tasks.ContinuationResultTaskFromResultTask<System.Threading.Tasks.Task[], System.Threading.Tasks.VoidTaskResult>.InnerInvoke() Line 213	C#
 	mscorlib.dll!System.Threading.Tasks.Task.Execute() Line 2498	C#
 	mscorlib.dll!System.Threading.Tasks.Task.ExecutionContextCallback(object obj) Line 2861	C#
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 954	C#
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 902	C#
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteWithThreadLocal(ref System.Threading.Tasks.Task currentTaskSlot) Line 2827	C#
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteEntry(bool bPreventDoubleExecution) Line 2756	C#
 	mscorlib.dll!System.Threading.Tasks.TaskScheduler.TryExecuteTask(System.Threading.Tasks.Task task) Line 458	C#
 	MonoDevelop.Core.dll!MonoDevelop.Core.SynchronizationContextTaskScheduler.PostCallback(object obj) Line 55	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.DispatchService.GtkSynchronizationContext.TimeoutProxy.HandlerInternal(System.IntPtr data) Line 77	C#
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	gtk-sharp.dll!Gtk.Application.Run()	Unknown
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeApp.Run() Line 404	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeStartup.Run(MonoDevelop.Ide.MonoDevelopOptions options) Line 310	C#
 	MonoDevelop.Ide.dll!MonoDevelop.Ide.IdeStartup.Main(string[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) Line 635	C#
 	MonoDevelop.exe!MonoDevelop.Startup.MonoDevelopMain.Main(string[] args) Line 39	C#
```

> VS bug [#623602](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/623602)

**Comment by KirillOsenkov** [comment]
I see an easy repro, will investigate.

**Comment by KirillOsenkov** [comment]
@minestarks this should be fixed in latest monodevelop master, let me know if you still see problems!

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4957 merged by: Kirill Osenkov  
 Author: mono/dev/kirillo  
 Commit: cc10c84ca1a98df7b415330027838b26c75d64f6 (xamarin/monodevelop)
